lib/usd/ui/layerEditor/layerTreeItem.cpp

```diff
@@ -363,22 +364,34 @@ void LayerTreeItem::saveAnonymousLayer()
 
     std::string fileName;
     if (sessionState->saveLayerUI(nullptr, &fileName)) {
+
+        std::string        filePath(fileName);
+        const std::string& extension = SdfFileFormat::GetFileExtension(filePath);
+        const std::string  defaultExt(UsdMayaTranslatorTokens->UsdFileExtensionDefault.GetText());
+        const std::string  usdCrateExt(UsdMayaTranslatorTokens->UsdFileExtensionCrate.GetText());
+        const std::string  usdASCIIExt(UsdMayaTranslatorTokens->UsdFileExtensionASCII.GetText());
+        if (extension != defaultExt && extension != usdCrateExt && extension != usdASCIIExt) {
+            filePath.append(".");
+            filePath.append(defaultExt.c_str());
+        }
```

Comment:
If this is the same code as in the other review, maybe create a function, something like ensureUSDFileExtension() or similar.

Reply:
done